src/apps/applib/lib/utility.c

```diff
@@ -355,9 +355,9 @@ void randomGeo(GeoCoord* g) {
 /**
  * Returns the number of non-invalid indexes in the array.
  */
-int countActualHexagons(H3Index* hexagons, int numHexagons) {
+int countActualHexagons(H3Index* hexagons, int numCells) {
```

Comment:
:+1:

Reply:
Hmm. This function is tricky. I suppose the *intention* is that it would only be used on arrays of cells, but it would also work for arrays of other modes (and would even include "invalid" indexes other than `H3_NULL` in the count).

And the "`Hexagons`" in the function name should at least be "`Cells`", if not the more general `"Indexes"`.

What would you all think about a function signature like:

```c
/**
 * Returns the number of indexes in the array which are not H3_NULL
 */
int countNonNullIndexes(H3Index* indexes, int numIndexes) {
```